### PR TITLE
53 docs base and thermo

### PR DESCRIPTION
**Pull request by jalew188**

### CODE REVIEW

File: alpharaw/ms_data_base.py [hunk 1]
```diff
@@ -6,7 +6,14 @@
 
 class MSData_Base:
     """
-    The base data structure for MS Data, other MSData loader inherit
+    The base data structure for MS Data, other MSData loader inherit this class.
```
Comment (mschwoer) [inline alpharaw/ms_data_base.py:9]: mb be more specific on what "MS Data" is here .. raw data I guess?

File: alpharaw/ms_data_base.py [hunk 2]
```diff
@@ -6,7 +6,14 @@
...
+
+    Parameters
+    ----------
+    centroided : bool, optional
```
Comment (mschwoer) [inline alpharaw/ms_data_base.py:13]: does this now mean we tell the data is centroided or that centroiding should be done?

File: alpharaw/ms_data_base.py [hunk 3]
```diff
@@ -25,7 +32,7 @@ class MSData_Base:
     """
     Spectrum dataframe containing the following columns:
 
-    - `rt` (float64): in minutes
+    - `rt` (float64): in minutes. `rt_sec` will be RT in seconds.
```
Comment (mschwoer) [inline alpharaw/ms_data_base.py:35]: ? `rt_sec` is not part of the interface?

File: alpharaw/ms_data_base.py [hunk 4]
```diff
@@ -62,15 +69,12 @@ class MSData_Base:
         "FT",
         "TOF",
     ]
+    """
+    Some spectrum infomation in str format that can be mapped into unique token IDs.
```
Comment (mschwoer) [inline alpharaw/ms_data_base.py:73]: mb give an example here?

File: alpharaw/ms_data_base.py [hunk 5]
```diff
@@ -82,9 +86,19 @@ def __init__(self, centroided: bool = True, save_as_hdf: bool = False, **kwargs)
...
-    def _get_term_id(self, terminology: str):
+    def _get_term_id(self, terminology: str) -> int:
         """
-        Get terminology id from :data:`self.vocab`, -1 if not exist.
+        Get terminology ID from :attr:`.MSData_Base.vocab`, -1 if not exist.
+
+        Parameters
+        ----------
+        terminology : str
+            The terminology name.
```
Comment (mschwoer) [inline alpharaw/ms_data_base.py:96]: mb give an example?

File: alpharaw/ms_data_base.py [hunk 6]
```diff
@@ -128,13 +158,29 @@ def _load_meta_from_hdf(self, hdf: HDF_File):
         self.instrument = hdf.ms_data.meta.instrument
 
     def save_hdf(self, _path: str):
+        """
+        Save data into HDF5 file
+
+        Parameters
+        ----------
+        _path : str
+            HDF5 file path
```
Comment (mschwoer) [inline alpharaw/ms_data_base.py:167]: in the past I learned that there's always a huge confusion in naming around files and paths.
So I made it a habit to be very precise, e.g. on `absolute` vs `relative` (also on `file` vs `file_name`)
Maybe rephrase to "absolute path to HDF5 file"

File: alpharaw/ms_data_base.py [hunk 7]
```diff
@@ -356,12 +419,37 @@ class MSReaderProvider:
...
         self.ms_reader_dict = {}
 
-    def register_reader(self, ms2_type: str, reader_class):
+    def register_reader(self, ms2_type: str, reader_class: type):
+        """
+        Register a new reader for `ms_type` format with `reader_class`.
+
+        Parameters
+        ----------
+        file_type : str
```
Comment (mschwoer) [inline alpharaw/ms_data_base.py:428]: does not match any of the parameters?

**Review by mschwoer — APPROVED** [review]
LGTM ..

File: alpharaw/ms_data_base.py (hunk 1 above)
Comment (jalew188) [reply alpharaw/ms_data_base.py:9]: We support not only raw data but also standard format like mzml. MB MS spectrum data?

File: alpharaw/ms_data_base.py (hunk 2 above)
Comment (jalew188) [reply alpharaw/ms_data_base.py:13]: MB centroiding is better?

File: alpharaw/ms_data_base.py (hunk 3 above)
Comment (jalew188) [reply alpharaw/ms_data_base.py:35]: No. Only `rt`

File: alpharaw/ms_data_base.py (hunk 2 above)
Comment (jalew188) [reply alpharaw/ms_data_base.py:13]: Anyway, I will change the name in other PRs, as this may affect alphadia.

File: alpharaw/ms_data_base.py [hunk 8]
```diff
@@ -100,15 +114,31 @@ def raw_file_path(self, _path: str):
         self._raw_file_path = _path
 
     def import_raw(self, _path: str):
```
[the remark this answers is not in the document]
Comment (jalew188) [reply alpharaw/ms_data_base.py:116]: Long history, I forgot to change it back

File: alpharaw/thermo.py [hunk 9]
```diff
@@ -14,6 +14,7 @@
     ms_reader_provider,
 )
 
+#: These thermo spectrum items can be only accessed by trailer dict using RawFileReader APIs.
```
[the remark this answers is not in the document]
Comment (jalew188) [reply alpharaw/thermo.py:17]: #: will be compiled by sphinx

File: alpharaw/thermo.py [hunk 10]
```diff
@@ -99,6 +91,19 @@ def _import(
         self,
         raw_file_path: str,
     ) -> dict:
+        """
+        Re-implementation of :func:`MSData_Base._import` to enable :func:`.MSData_Base.import_raw`.
+
+        Parameters
+        ----------
+        raw_file_path : str
+            File path of the raw data.
```
[the remark this answers is not in the document]
Comment (mschwoer) [reply alpharaw/thermo.py:100]: fair enough, then mention "absolute or relative path" ..
